### PR TITLE
Enhance the OpenId settings name and descriptions

### PR DESCRIPTION
**Pull request by MikeAlhayek**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs [hunk 1]
```diff
@@ -19,18 +19,6 @@ public int Create()
             var S = shellScope.ServiceProvider.GetService<IStringLocalizer<DefaultScopesMigration>>();
             var scopeManager = shellScope.ServiceProvider.GetRequiredService<IOpenIdScopeManager>();
 
-            if (await scopeManager.FindByNameAsync(OpenIddictConstants.Scopes.Profile) == null)
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs:22]: Out of curiosity, why this change?

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml [hunk 2]
```diff
@@ -13,209 +14,210 @@
     <label asp-for="DisplayName" class="form-label">@T["Display Name"]</label>
     <input asp-for="DisplayName" class="form-control" />
     <span asp-validation-for="DisplayName"></span>
-    <span class="hint">@T["Display name of the IdP"]</span>
+    <span class="hint">@T["User-Friendly Display Name of the Identity Provider"]</span>
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:17]: I'd personally avoid using upper-casing in descriptions or hints.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml [hunk 3]
```diff
@@ -13,209 +14,210 @@
...
 
 <div class="mb-3" asp-validation-class-for="SignedOutCallbackPath">
     <label asp-for="SignedOutCallbackPath" class="form-label">@T["SignedOut CallbackPath"]</label>
     <input asp-for="SignedOutCallbackPath" placeholder="/signout-callback-oidc" class="form-control" />
     <span asp-validation-for="SignedOutCallbackPath"></span>
-    <span class="hint">@T["The request path within the application's base path where the user agent will be returned after sign out from the identity provider."]</span>
-    <a class="hint" href="http://openid.net/specs/openid-connect-session-1_0.html#RedirectionAfterLogout" target="_blank">@T["See post_logout_redirect_uri from "]http://openid.net/specs/openid-connect-session-1_0.html#RedirectionAfterLogout</a>
+    <span class="hint">@T["The request path within the application's base path where the user-agent will be redirected after signing out from the identity provider."]</span>
+    <span class="hint">
+        @T["To see 'post_logout_redirect_uri' from, visit: <a href=\"{0}\" target=\"_blank\">", "http://openid.net/specs/openid-connect-session-1_0.html#RedirectionAfterLogout"]</a>
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:47]: Sounds grammatically weird, no?

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml [hunk 4]
```diff
@@ -13,209 +14,210 @@
...
-        <option id="@OpenIdConnectResponseMode.Query" value="@OpenIdConnectResponseMode.Query">@T[OpenIdConnectResponseMode.Query]</option>
-    </select>
-    <span asp-validation-for="ResponseMode"></span>
-    <span class="hint">@T["Configure Response Mode see: http://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthResponse"]</span>
-</div>
+    <div class="mb-3" asp-validation-class-for="ResponseMode">
+        <label asp-for="ResponseMode" class="form-label">@T["Response Mode"]</label>
+        <select asp-for="ResponseMode" class="form-select">
+            <option id="@OpenIdConnectResponseMode.FormPost" value="@OpenIdConnectResponseMode.FormPost">@T[OpenIdConnectResponseMode.FormPost]</option>
+            @*<option id="@OpenIdConnectResponseMode.Fragment" value="@OpenIdConnectResponseMode.Fragment">@T[OpenIdConnectResponseMode.Fragment]</option>*@
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:146]: Let's remove it, it's not something we'll ever support.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs (hunk 1 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs:22]: Because `openid` and `profile` scopes are always included with every request. So I don't think we need the user to explicitly including these 2.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml (hunk 3 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:47]: when I read it again yes. Because I read `form` instead of `from` :). I'll fix it.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml (hunk 2 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:17]: "Display Name" and "Identity Provider" are nouns and why they are capitalized in this case.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml (hunk 4 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: done.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml (hunk 3 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: fixed

File: src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs (hunk 1 above)
Comment (kevinchalet) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs:22]: Not all OIDC clients request the `profile` scope. And in a (distant) future, we'll want to update the consent view to display the scope description (and potentially allow users to deselect some of the requested scopes).

I'd keep it.

Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs:22]: done.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs [hunk 5]
```diff
@@ -19,6 +19,18 @@ public int Create()
             var S = shellScope.ServiceProvider.GetService<IStringLocalizer<DefaultScopesMigration>>();
             var scopeManager = shellScope.ServiceProvider.GetRequiredService<IOpenIdScopeManager>();
 
+            if (await scopeManager.FindByNameAsync(OpenIddictConstants.Scopes.Email) == null)
+            {
+                var descriptor = new OpenIdScopeDescriptor
+                {
+                    DisplayName = S["Email"],
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs:26]: Hum, I missed in the previous that that we're localizing the display name and the description. It's not wrong, but these fields will be populated using the locale used by the user when setting up the OpenID module, which may lead to unwanted results.

Note: OpenIddict has dedicated `LocalizedDisplayName` and `LocalizedDescriptions` properties that allow storing multiple localized versions (it's a dictionary using `CultureInfo` as the key).

**Review by kevinchalet — APPROVED** [review]

File: src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs (hunk 5 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Migrations/DefaultScopesMigration.cs]: removed localization
